packages/orbit-components/src/Badge/index.tsx

```diff
@@ -1,46 +1,46 @@
 "use client";
 
 import React from "react";
+import clsx from "clsx";
 
 import BadgePrimitive from "../primitives/BadgePrimitive";
 import { TYPE_OPTIONS, TOKENS } from "./consts";
-import useTheme from "../hooks/useTheme";
-import type { Theme } from "../defaultTheme";
 import type { Props, Type } from "./types";
 
-const getTypeToken = ({ name, theme, type }: { name: string; theme: Theme; type: Type }) => {
+const getTypeToken = ({ name, type }: { name: string; type: Type }) => {
   const tokens = {
     [TOKENS.background]: {
-      [TYPE_OPTIONS.NEUTRAL]: theme.orbit.badgeNeutralBackground,
-      [TYPE_OPTIONS.INFO_SUBTLE]: theme.orbit.badgeInfoSubtleBackground,
-      [TYPE_OPTIONS.SUCCESS_SUBTLE]: theme.orbit.badgeSuccessSubtleBackground,
-      [TYPE_OPTIONS.WARNING_SUBTLE]: theme.orbit.badgeWarningSubtleBackground,
-      [TYPE_OPTIONS.CRITICAL_SUBTLE]: theme.orbit.badgeCriticalSubtleBackground,
-      [TYPE_OPTIONS.DARK]: theme.orbit.badgeDarkBackground,
-      [TYPE_OPTIONS.WHITE]: theme.orbit.badgeWhiteBackground,
-      [TYPE_OPTIONS.INFO]: theme.orbit.badgeInfoBackground,
-      [TYPE_OPTIONS.CRITICAL]: theme.orbit.badgeCriticalBackground,
-      [TYPE_OPTIONS.SUCCESS]: theme.orbit.badgeSuccessBackground,
-      [TYPE_OPTIONS.WARNING]: theme.orbit.badgeWarningBackground,
-      [TYPE_OPTIONS.BUNDLE_BASIC]: theme.orbit.badgeBundleBasicBackground,
-      [TYPE_OPTIONS.BUNDLE_MEDIUM]: theme.orbit.badgeBundleMediumBackground,
-      [TYPE_OPTIONS.BUNDLE_TOP]: theme.orbit.badgeBundleTopBackground,
+      [TYPE_OPTIONS.NEUTRAL]: "bg-badge-neutral-background",
+      [TYPE_OPTIONS.INFO_SUBTLE]: "bg-badge-info-subtle-background",
+      [TYPE_OPTIONS.SUCCESS_SUBTLE]: "bg-badge-success-subtle-background",
+      [TYPE_OPTIONS.WARNING_SUBTLE]: "bg-badge-warning-subtle-background",
+      [TYPE_OPTIONS.CRITICAL_SUBTLE]: "bg-badge-critical-subtle-background",
+      [TYPE_OPTIONS.DARK]: "bg-badge-dark-background",
+      [TYPE_OPTIONS.WHITE]: "bg-badge-white-background",
+      [TYPE_OPTIONS.INFO]: "bg-badge-info-background",
+      [TYPE_OPTIONS.CRITICAL]: "bg-badge-critical-background",
+      [TYPE_OPTIONS.SUCCESS]: "bg-badge-success-background",
+      [TYPE_OPTIONS.WARNING]: "bg-badge-warning-background",
+      // TODO: repalce with proper granular tokens for gradients
+      [TYPE_OPTIONS.BUNDLE_BASIC]: "bg-badge-bundle-basic-background",
+      [TYPE_OPTIONS.BUNDLE_MEDIUM]: "bg-badge-bundle-medium-background",
+      [TYPE_OPTIONS.BUNDLE_TOP]: "bg-badge-bundle-top-background",
```

Comment:
These are not working. For the same reason presented in #4015

Reply:
I've left the comment above and in PR description